packages/oauth/src/index.spec.js

```diff
@@ -0,0 +1,357 @@
+    it('should fail if database does not have an entry for authorize query', async () => {
+      const installer = new InstallProvider({clientId, clientSecret, stateSecret, installationStore});
+      try {
+          const AuthResult = await installer.authorize({teamId:'teamID'});
+      } catch(error) {
+          assert.equal(error.code, 'INSTALLER_AUTHORIZATION_ERROR');
+          assert.equal(error.message, 'Failed fetching data from the Installation Store');
+      }
+    });
+
+    it('should successfully return the Installation Object from the database', async () => {
+      const installer = new InstallProvider({clientId, clientSecret, stateSecret, installationStore});
+      const fakeAuthResult = {
+        userToken: 'userToken',
+        botToken: 'botToken',
+        botId: 'botId',
+        botUserId: 'botUserId'
+      };
+
+      try {
+        const AuthResult = await installer.authorize({teamId:'test-team-id'});
+        assert.equal(AuthResult.userToken, fakeAuthResult.userToken);
+        assert.equal(AuthResult.botToken, fakeAuthResult.botToken);
+        assert.equal(AuthResult.botId, fakeAuthResult.botId);
+        assert.equal(AuthResult.botUserId, fakeAuthResult.botUserId);
+      } catch(error) {
+        assert.fail(error.message);
+      }
+    });
+  });
+
+  describe('installer.handleCallback', async () => {
+    it('should call the failure callback due to missing state query parameter on the URL', async () => {
+      const req = { url: 'someUrl' };
+      let sent = false;
+      const res = { send: () => { sent = true; } };
+      const callbackOptions = {
+        success: async (installation, installOptions, req, res) => {
+          res.send('successful!');
+          assert.fail('should have failed');
+        },
+        failure: async (error, installOptions , req, res) => {
+          assert.equal(error.code, 'MISSING_STATE')
+          res.send('failure');
+        },
+      }
+      const installer = new InstallProvider({clientId, clientSecret, stateSecret, installationStore, logger: noopLogger});
+      await installer.handleCallback(req, res, callbackOptions);
+
+      assert.isTrue(sent);
+    });
+    it('should call the success callback for a v2 url', async () => {
+      let sent = false;
+      const installUrlOptions = { scopes: [ 'channels:read' ] };
+      const res = { send: () => { sent = true; } };
+      const callbackOptions = {
+        success: async (installation, installOptions, req, res) => {
+          assert.equal(installOptions.scopes[0], installUrlOptions.scopes[0]);
+          res.send('successful!');
+        },
+        failure: async (error, installOptions , req, res) => {
+          assert.fail(error.message);
+          res.send('failure');
+        },
+      }
+      
+      const installer = new InstallProvider({clientId, clientSecret, stateSecret, installationStore});
+      const url = await installer.generateInstallUrl(installUrlOptions);
+      const req = { url };
```

Comment:
initialize the installprovider with a fake state store

```suggestion
      const fakeState = Symbol();
      const fakeCode = Symbol();
      const req = { url: `http://example.com?state=${fakeState}&code=${fakeCode}` };
```

then check 1) that the fake state store only resolves the promise when the state matches fakeState, and 2) the sinon fake is called with the fakeCode.


this is not required, just could be an improvement.